impl/src/main/java/org/jboss/weld/bootstrap/BeanDeployer.java

```diff
@@ -122,7 +122,8 @@ private void processPriority(AnnotatedType<?> type) {
 
     public <T> BeanDeployer addSyntheticClass(AnnotatedType<T> source, Extension extension, String suffix) {
         if (suffix == null) {
-            suffix = AnnotatedTypes.createTypeId(source);
+            // If there is a synthetic AnnotatedType added for the given class then calculate the suffix, otherwise use extension FQCN
+            suffix = getEnvironment().hasSyntheticAnnotatedType(source.getJavaClass()) ? AnnotatedTypes.createTypeId(source) : extension.getClass().getName();
```

Comment:
This is actually wrong as it makes the id non-deterministic (if multiple ATs for a given class exist).